### PR TITLE
TCN-667 add release action

### PR DESCRIPTION
**Pull request by elliotmjackson**
```sh
cd "${BUFBUILD_BUF_SETUP_ACTION_DIR}"
git switch main
git pull origin main

make updateversion VERSION=${VERSION}
git diff

git checkout -b release/v${VERSION}
git commit -am "Update to v${VERSION}"
git push origin release/v${VERSION}

# Get that PR approved and merged

git switch main
git pull origin main
git tag -a "v${VERSION}" -m "v${VERSION}"
git push origin "v${VERSION}"
```

Resolves TCN-680
Resolves [TCN-680](https://linear.app/bufbuild/issue/TCN-680)

### CODE REVIEW

**Comment by ElliotMJackson** [comment]
depends on TCN-671

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          token: ${{ steps.generate_token.outputs.token }}
+  tag:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request' && github.event.pull_request.merged == true && startsWith(github.event.pull_request.head.ref, 'release') }}
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: 251311
```
Comment (ElliotMJackson) [inline .github/workflows/release.yaml:54]:
```suggestion
          app_id: 257262
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        description: The version you intend to release (eg x.y.z)
+
+env:
+  VERSION: v${{ github.event.inputs.version }}
+  APP_ID: 257262
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.version != '' }}
```
[the remark this answers is not in the document]
Comment (elliotmjackson) [reply .github/workflows/release.yaml:16]:
```suggestion
    if: ${{ github.event.inputs.version != '' }}
```

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+name: Release
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        description: The version you intend to release (eg x.y.z)
```
Comment (elliotmjackson) [inline .github/workflows/release.yaml:7]:
```suggestion
        description: The version you intend to release (eg x.y.z)
        required: true
```

**Pull request merged**